airflow/jobs/scheduler_job_runner.py

```diff
@@ -348,6 +348,7 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session) -
 
         # dag_id to # of running tasks and (dag_id, task_id) to # of running tasks.
         concurrency_map = self.__get_concurrency_maps(states=EXECUTION_STATES, session=session)
+        concurrency_map_deferred = self.__get_concurrency_maps(states=DEFERRED_STATES, session=session)
```

Comment:
This should be something like

```python
if max_active_tasks_include_deferred:
    concurrency_map = self.__get_concurrency_maps(states=DEFERRED_STATES | EXECUTION_STATE, session=session)
else:
    concurrency_map = self.__get_concurrency_maps(states=EXECUTION_STATES, session=session)
```

to save one db query.

Reply:
Yes agree to save one db query , but 
I believe `max_active_tasks_include_deferred` config is accessible only inside for loop from the task_instances_to_examine https://github.com/apache/airflow/blob/42a1ae601b4f7d46299e1ecde0aa0e403f533739/airflow/jobs/scheduler_job_runner.py#L435, not sure how to update your suggestion in here at top level. happy to take some suggestion , please let me know. :)

Reply:
@uranusjr Can i have review please, is the updates are fine :)

Reply:
I'm not quite sure how to apply the suggested approach because max_active_tasks_include_deferred is a DAG-level setting. A user might enable this option for only a few DAGs. Therefore, if we send states=DEFERRED_STATES | EXECUTION_STATE to __get_concurrency_maps, it could retrieve all tasks which are in deferred state, even for DAGs that do not have max_active_tasks_include_deferred enabled.

I have updated `__get_concurrency_maps` to check for max_active_tasks_include_deferred. i believe which is one query but extra join condition.